### PR TITLE
[CALCITE-5884] ARRAY_TO_STRING function should return NULL if its 'nullValue' argument is NULL

### PR DESCRIPTION
**Pull request by mihaibudiu**

### CODE REVIEW

**Comment by mihaibudiu** [comment]
@zoudan I think you have implemented this function.

**Comment by mihaibudiu** [comment]
I would appreciate if someone could review this PR.

**Comment by LakeShen** [comment]
@mihaibudiu ,please rebase the latest main branch's code.

**Comment by mihaibudiu** [comment]
> @mihaibudiu ,please rebase the latest main branch's code.

Done

**Review by julianhyde — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java [hunk 1]
```diff
@@ -1925,6 +1925,18 @@ private void checkSemiOrAntiJoinProjectTranspose(JoinRelType type) {
         .check();
   }
 
+  /** Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-5884">[CALCITE-5884]
+   * Description of ARRAY_TO_STRING function is incomplete</a>. */
```
[the remark this answers is not in the document]
Comment (mihaibudiu) [reply core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java]: There were more instances, I fixed all of them in this file.

File: core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java [hunk 2]
```diff
@@ -5471,6 +5471,10 @@ public static String arrayToString(List list, String delimiter) {
 
   /** SQL {@code ARRAY_TO_STRING(array, delimiter, nullText)} function. */
   public static String arrayToString(List list, String delimiter, @Nullable String nullText) {
+    // Note that the SQL function ARRAY_TO_STRING that we implement will return
+    // 'NULL' when the nullText argument is NULL.  However, that is handled by
+    // the nullPolicy of the RexToLixTranslator.  So here a NULL value
```
Comment (libenchao) [inline core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java:5476]: NIT: redundant spaces (and in above line too)

**Review by libenchao — COMMENTED** [review]
Thanks @mihaibudiu, it looks good to me with only nit comments. Please note that Julian has proposed a better summary in Jira.

File: core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java (hunk 2 above)
Comment (mihaibudiu) [reply core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java]: I have changed the summary, removed the spaces, and squashed the commits.

**Review by libenchao — APPROVED** [review]
+1

**Comment by mihaibudiu** [comment]
@julianhyde I hope I have made the changes you requested

**Comment by libenchao** [comment]
Can we move on with merging, or we can leave it out of 1.36.0? I plan to start the release process if there is still no progress on this PR in a day  :)

**Comment by mihaibudiu** [comment]
I think we should just merge it. I think it's a positive improvement in documentation.

**Pull request merged**